### PR TITLE
Withdraw invalidated claims

### LINKED ISSUE #674: Non-Fill: Withdraw claims when L1 resolved Non-Fill arrives
Once a L1 transmitted non-fill proof arrives, the agent needs to decide whether or not to withdraw a claim.

### Tasks
- [ ] Check if `withdraw` should be triggered when a claim is in the `invalidated_l1_resolved` state
- [ ] Implement necessary changes to `maybe_withdraw`
- [ ] Add tests

### PR DESCRIPTION
**Pull request by palango**
Resolves #674 
Resolves https://github.com/beamer-bridge/beamer/issues/547

### CODE REVIEW

File: beamer/chain.py [hunk 1]
```diff
@@ -459,6 +459,9 @@ def maybe_withdraw(claim: Claim, context: Context) -> None:
         if context.address in winning_addresses:
             _withdraw(claim, context)
 
+    if claim.is_invalidated_l1_resolved and agent_is_challenger:
+        _withdraw(claim, context)
+
```
Comment (fredo) [inline beamer/chain.py:464]: I think we have to include that into the `if / elif condition`. Equal to the `has_l1_resolution` the timestamp doesn't matter if the fill hash is invalidated. So it should be prioritized higher than the claim termination.
In addition to that I think it is possible that we call `_withdraw()` twice the way it is right now.

File: beamer/tests/agent/unit/test_invalidate.py [hunk 2]
```diff
@@ -159,3 +163,39 @@ def test_maybe_invalidate_claim_wrong_fill_id_but_timed_out(mocked_invalidate):
...
+    ],
+)
+def test_maybe_withdraw_after_invalidation(mocked_withdraw, test_data):
+    context, _ = make_context()
+
+    request = make_request()
+    context.requests.add(request.id, request)
+
+    participant1 = test_data[0]
+    participant2 = test_data[1]
```
Comment (fredo) [inline beamer/tests/agent/unit/test_invalidate.py:184]: renaming it to claimer and challenger would be more explicit

Comment (palango) [reply beamer/tests/agent/unit/test_invalidate.py:184]: ✔️

File: beamer/tests/agent/unit/test_invalidate.py [hunk 3]
```diff
@@ -159,3 +163,39 @@ def test_maybe_invalidate_claim_wrong_fill_id_but_timed_out(mocked_invalidate):
...
+
+    claim = make_claim_challenged(
+        request=request,
+        claimer=participant1,
+        challenger=participant2,
+        # Claim should not be withdrawable yet
+        termination=Termination(TIMESTAMP + 1),
+    )
+    context.claims.add(claim.id, claim)
+    claim.start_challenge(make_tx_hash())
```
[the remark this answers is not in the document]
Comment (palango) [reply beamer/tests/agent/unit/test_invalidate.py:195]: ✔️

File: beamer/chain.py (hunk 1 above)
Comment (palango) [reply beamer/chain.py]: ✔️

**Review by fredo — APPROVED** [review]
LGTM!

**Pull request merged**